Remove failing user ID sanity check on DPS Payment callback

markhowardnz opened the pull request:
As discussed in email. 

Added more logging to DPS callback to try and clarify which failure cases are *actually* happening in response to DPS callbacks.

Removed user_id sanity check in DPS callback, as it seems to be kicking out legitimate users if they call out to DPS from one node, but are called back after payment and happen to land on the other node.

Commit:
Added more logging to DPS callback to try and clarify which failure cases are actually happening in response to DPS callbacks.
Removed user_id sanity check in DPS callback, as it seems to be kicking out legitimate users.

gbentley reviewed (commented):
@markhowardnz Have you thought about adding an extra check whether the current ID is the AnonymousUserID, so we retain the existing functionality for logged-in users else use add logging as per your changes?

Commit:
Refined user_id sanity check to block users if a purchase was made with a logged in user, and the currently logged in user is not the same person as placed the order.
Allow through the case where the current user is Anonymous, but the order's user is not. This may happen (ahem) legitimately if the session/user data is lost in redirection.

markhowardnz commented:
I've updated the check - is that what you had in mind?

Should now always log a message containing the user IDs (based on the fact that logging this is helpful for troubleshooting). 

Will now stop users if they don't match (ie logged in users != expected ordering user), but only if the current user is not Anonymous. The situation where a logged in user becomes anonymous at redirection time seems to be the root cause of the problem. This check shouldn't fail this case, although it will log a message to explain what happened.

gbentley commented:
Yep, perfect.

gbentley commented:
Work halted due to request from Leo.
